includes/components/class-user.php

```diff
@@ -62,11 +62,46 @@ public function __construct( $args = [] ) {
 
 			// Alter templates.
 			add_filter( 'hivepress/v1/templates/site_footer_block', [ $this, 'alter_site_footer_block' ] );
+
+			// Update last seen.
+			add_action( 'wp_footer', [ $this, 'update_last_seen' ] );
 		}
 
 		parent::__construct( $args );
 	}
 
+	/**
+	 * Updates user last seen.
+	 */
+	public function update_last_seen() {
```

Comment:
This "online" indication feature should also be optional, please add an option to HivePress/Settings/Users/Display, e.g. "Indicate online status", which should add this dot with "last seen", without this option enabled no related functions should run (e.g. add a guard clause inside the callback at the beginning)

Reply:
Another issue is once we add real-time messaging then user can stay on the same page without refreshing it, the same issue would occur if AJAX search result refreshing is added, a possible solution is refreshing "last seen" in 2 cases instead of rendering wp_footer:

- request context (runs on every front-end page)
- REST API (when the logged in user makes any request to the REST API, refresh "last seen")